source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java

```diff
@@ -43,23 +42,21 @@
   public boolean isItDead(final boolean runningWorkers, final int numberOfMessages) {
     boolean doReset = false;
     if (runningWorkers && numberOfMessages == 0) {
-      if (firstProblem == 0) {
-        firstProblem = new Date().getTime();
+      if (firstProblem == null) {
+        firstProblem = now().plusMinutes(RESET_TIME_MINUTES);
```

Comment:
I think it's still cleaner to do the check in the `else` like before.
Now we suggest that the first problem happened 10 minutes in the future.
Is there a reason this was moved to the `if`-statement?

Reply:
It doesn't do the check here, but sets the time to measure to the time it should trigger. But it doesn't reflect the naming so I've moved it to the else part.